server/task/src/main/java/org/apache/james/task/MemoryTaskManagerWorker.java

```diff
@@ -88,8 +87,8 @@ public void cancelTask(TaskId id, Consumer<TaskExecutionDetailsUpdater> updateDe
         Optional.ofNullable(idToFuture.remove(id))
             .ifPresent(future -> {
                 requestCancellation(updateDetails, future);
-                waitUntilFutureIsCancelled(future)
-                    .subscribe(cancellationSuccessful -> effectivelyCancelled(updateDetails));
+                waitUntilFutureIsCancelled(future).blockFirst();
```

Comment:
I guess that's the only real change, right?

Reply:
No. (cf above)